### PR TITLE
560 update existing email content with mention of references

### PR DESCRIPTION
**Pull request by JonF1**
## Context
We have a number of existing emails that need to be updated to inform candidates that references form part of an offer condition.

## Changes proposed in this pull request
* Update copy in various emails but behind a feature flag as providers are still able to make offers to candidates in the old cycle for a couple of weeks

## Guidance to review

Commit by commit. Email previews at `/rails/mailers/candidate_mailer/`

## Link to Trello card
https://trello.com/c/EL6GilJG/560-update-existing-email-content-with-mention-of-references

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/views/candidate_mailer/withdraw_last_application_choice.text.erb [hunk 1]
```diff
@@ -3,3 +3,13 @@ Dear <%= @application_form.first_name %>
 You’ve withdrawn your <%= 'application'.pluralize(@withdrawn_course_names.size) %> for <%= @withdrawn_course_names.to_sentence %>.
 
 <%= render 'still_interested_content' %>
+
+<% if @application_form.show_new_reference_flow? %>
+Since you withdrew your application, we’ve contacted these people to say they do not need to give a reference:
```
Comment (JonF1) [inline app/views/candidate_mailer/withdraw_last_application_choice.text.erb:8]: This is in the new copy but I can't see where this happens in the withdraw application service

File: app/views/candidate_mailer/withdraw_last_application_choice.text.erb [hunk 2]
```diff
@@ -3,3 +3,13 @@ Dear <%= @application_form.first_name %>
...
+
+<% @application_form.application_references.select(&:feedback_requested?).each do |reference| %>
```
Comment (carlosmartinez) [inline app/views/candidate_mailer/withdraw_last_application_choice.text.erb:10]: Rails magic gives you `@application_form.application_references.feedback_requested`

File: spec/mailers/candidate_mailer_spec.rb [hunk 3]
```diff
@@ -199,25 +199,51 @@
   end
 
   describe '.withdraw_last_application_choice' do
-    let(:email) { mailer.withdraw_last_application_choice(application_form) }
+    let(:application_form_with_references) do
+      create(:application_form, first_name: 'Fred',
```
Comment (carlosmartinez) [inline spec/mailers/candidate_mailer_spec.rb:203]: how about
```
let(:recruitment_cycle_year) { ApplicationForm::OLD_REFERENCE_FLOW_CYCLE_YEAR }
let(:application_form_with_references) do
      create(:application_form, first_name: 'Fred',
                                recruitment_cycle_year: recruitment_cycle_year,
                                application_choices: application_choices,
                                application_references: [referee1, referee2])
    end
```

File: spec/mailers/candidate_mailer_spec.rb [hunk 4]
```diff
@@ -199,25 +199,51 @@
...
     end
 
-    context 'when a candidate has 2 or 3 offers that were declined' do
-      let(:application_choices) { [build_stubbed(:application_choice, :withdrawn), build_stubbed(:application_choice, :withdrawn)] }
+    context 'when new reference flow is active' do
+      let(:application_choices) { [create(:application_choice, status: 'withdrawn')] }
+
+      before do
+        FeatureFlag.activate(:new_references_flow)
+        application_form_with_references.recruitment_cycle_year = 2023
```
Comment (carlosmartinez) [inline spec/mailers/candidate_mailer_spec.rb:227]: and then you can remove this

File: spec/mailers/candidate_mailer_spec.rb [hunk 5]
```diff
@@ -199,25 +199,51 @@
...
+        'You’ve withdrawn your application',
         'heading' => 'Dear Fred',
         'application_withdrawn' => 'You’ve withdrawn your application',
       )
     end
 
-    context 'when a candidate has 2 or 3 offers that were declined' do
-      let(:application_choices) { [build_stubbed(:application_choice, :withdrawn), build_stubbed(:application_choice, :withdrawn)] }
+    context 'when new reference flow is active' do
+      let(:application_choices) { [create(:application_choice, status: 'withdrawn')] }
```
Comment (carlosmartinez) [inline spec/mailers/candidate_mailer_spec.rb:223]: let(:recruitment_cycle_year) { ApplicationForm::OLD_REFERENCE_FLOW_CYCLE_YEAR + 1 }

File: spec/mailers/candidate_mailer_spec.rb [hunk 6]
```diff
@@ -364,6 +364,17 @@
       'pending condition text' => 'You still need to meet the following condition',
       'pending condition' => 'Be cool',
     )
+
+    context 'when new reference flow is active' do
+      before do
+        FeatureFlag.activate(:new_references_flow)
+        application_form.recruitment_cycle_year = 2023
```
Comment (carlosmartinez) [inline spec/mailers/candidate_mailer_spec.rb:397]: similarly, could this be shifted into the declaration of `application_form` and the magic number replaced?

File: app/views/candidate_mailer/offer_accepted.text.erb [hunk 7]
```diff
@@ -2,6 +2,24 @@ Dear <%= @application_form.first_name %>
 
 You’ve accepted <%= @provider_name %>’s offer to study <%= @course_name_and_code %>.
 
+<% if @application_form.show_new_reference_flow? %>
+  
+  Your place will be confirmed when:
+  - you’ve met your offer conditions
+  - when safeguarding checks like DBS and references have been completed
```
Comment (carlosmartinez) [inline app/views/candidate_mailer/offer_accepted.text.erb:9]: "when" is duplicated

File: spec/mailers/candidate_mailer_spec.rb [hunk 8]
```diff
@@ -300,6 +300,18 @@
       'offer_details' => 'You’ve accepted Arithmetic College’s offer to study Mathematics (M101)',
       'course start' => 'September 2021',
     )
+
+    context 'when new reference flow is active' do
+      before do
+        FeatureFlag.activate(:new_references_flow)
+        application_form.recruitment_cycle_year = 2023
```
Comment (carlosmartinez) [inline spec/mailers/candidate_mailer_spec.rb:333]: as previous, apols!

File: spec/mailers/previews/candidate_mailer_preview.rb [hunk 9]
```diff
@@ -858,6 +859,10 @@ def nudge_unsubmitted_with_incomplete_references
   end
 
 private
+  def new_references_content(application_form)
+    FeatureFlag.activate(:new_references_flow)
+    application_form.recruitment_cycle_year = 2023
```
Comment (carlosmartinez) [inline spec/mailers/previews/candidate_mailer_preview.rb:872]: i don't make the rules

**Comment by carlosmartinez** [comment]
all looking good, just some minor comments

File: spec/mailers/candidate_mailer_spec.rb (hunk 3 above)
Comment (JonF1) [reply spec/mailers/candidate_mailer_spec.rb:206]: thanks, done

File: app/views/candidate_mailer/offer_accepted.text.erb (hunk 7 above)
Comment (JonF1) [reply app/views/candidate_mailer/offer_accepted.text.erb]: good spot

**Review by JR-G — APPROVED** [review]

**Comment by JR-G** [comment]
Have approved following on from the comments from Carlos so 🚢  once done 🙌

**Pull request merged**